### PR TITLE
New string_add_assign lint 

### PR DESCRIPTION
**Pull request by llogiq**
This is the first part of #121, also I reformatted and refactored the source a bit.

### CODE REVIEW

File: src/strings.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+#[derive(Copy,Clone)]
+pub struct StringAdd;
+
+impl LintPass for StringAdd {
+    fn get_lints(&self) -> LintArray {
+        lint_array!(STRING_ADD_ASSIGN)
+    }
+    
+    fn check_expr(&mut self, cx: &Context, e: &Expr) {
+        if let &ExprAssign(ref target, ref  src) =&e.node {
```
Comment (Manishearth) [inline src/strings.rs]: nit: ws

File: src/strings.rs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            if is_string(cx, target) && is_add(src, target) { 
+                span_lint(cx, STRING_ADD_ASSIGN, e.span, 
+                    "You assign the result of adding something to this string. \
```
Comment (Manishearth) [inline src/strings.rs]: Shouldn't we just check for `s = s + y` here, not `s = x + y`?

**Comment by llogiq** [comment]
We actually check for `s = s + ..`: At line 49, I use `eq_op::is_exp_eq` to do this (which is why I made it `pub`lic.

**Comment by Manishearth** [comment]
Cool, r+

**Pull request merged**